packages/utils/src/internals/sitemap.ts

```diff
@@ -1,36 +1,44 @@
-import { Readable, Writable } from 'node:stream';
+import { createHash } from 'node:crypto';
+import type { Duplex} from 'node:stream';
+import { PassThrough, Readable, Transform, pipeline } from 'node:stream';
 import { StringDecoder } from 'node:string_decoder';
 import { createGunzip } from 'node:zlib';
 
 import log from '@apify/log';
-import type { SAXStream } from 'sax';
 import sax from 'sax';
 import MIMEType from 'whatwg-mimetype';
 
-type SitemapSource = { type: 'url'; url: string } | { type: 'raw'; content: string };
-
-class ParsingState {
-    sources: SitemapSource[] = [];
-    urls: string[] = [];
-    visitedSitemapUrls: string[] = [];
-    context?: 'sitemapindex' | 'urlset';
-    loc = false;
 
-    resetContext() {
-        this.context = undefined;
-        this.loc = false;
-    }
+interface SitemapUrlData {
+    url: string;
```

Comment:
I know this is 1000% more user friendly, but in sitemaps, the URL of the page is in the `<loc>` element... `<url>` is the parent element 

![obrazek](https://github.com/apify/crawlee/assets/61918049/9cf574b7-88a0-4eeb-b8c4-f0d251dcdc6f)

Reply:
I'd call that a minor weirdness of the xml schema and ignore it. What do you suggest?

Reply:
Well, given we're following the naming with all the other elements, I would go with `loc` instead of `url`... but I'm not willing to die on this hill 🤷🏽